All tests that contain inner classes will be skipped if triggered from the test explorer

Result:
![image](https://user-images.githubusercontent.com/22470312/49061813-c321ab00-f24c-11e8-910c-449891fa9ea5.png)

Output
```
[ERROR] @@<TestRunner-{"name":"error", "attributes":{"message":"Failed to parse tests for suite com.example.project.JUnit4Test$A#testAssert1. Details: com.example.project.JUnit4Test$A#testAssert1.", "details":"java.lang.ClassNotFoundException: com.example.project.JUnit4Test$A#testAssert1\r\n\tat java.net.URLClassLoader.findClass(URLClassLoader.java:381)\r\n\tat java.lang.ClassLoader.loadClass(ClassLoader.java:424)\r\n\tat sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)\r\n\tat java.lang.ClassLoader.loadClass(ClassLoader.java:357)\r\n\tat java.lang.Class.forName0(Native Method)\r\n\tat java.lang.Class.forName(Class.java:264)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4RunnerUtil.getRequestForClasses(JUnit4RunnerUtil.java:80)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4RunnerUtil.createTestReferences(JUnit4RunnerUtil.java:45)\r\n\tat com.microsoft.java.test.runner.junit4.CustomizedJUnit4CoreRunner.run(CustomizedJUnit4CoreRunner.java:25)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4Launcher.execute(JUnit4Launcher.java:30)\r\n\tat com.microsoft.java.test.runner.Launcher.main(Launcher.java:53)\r\n"}}-TestRunner>.
[ERROR] 
.
[ERROR] @@<TestRunner-{"name":"error", "attributes":{"message":"Failed to parse tests for suite com.example.project.JUnit4Test#test. Details: com.example.project.JUnit4Test#test.", "details":"java.lang.ClassNotFoundException: com.example.project.JUnit4Test#test\r\n\tat java.net.URLClassLoader.findClass(URLClassLoader.java:381)\r\n\tat java.lang.ClassLoader.loadClass(ClassLoader.java:424)\r\n\tat sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)\r\n\tat java.lang.ClassLoader.loadClass(ClassLoader.java:357)\r\n\tat java.lang.Class.forName0(Native Method)\r\n\tat java.lang.Class.forName(Class.java:264)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4RunnerUtil.getRequestForClasses(JUnit4RunnerUtil.java:80)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4RunnerUtil.createTestReferences(JUnit4RunnerUtil.java:45)\r\n\tat com.microsoft.java.test.runner.junit4.CustomizedJUnit4CoreRunner.run(CustomizedJUnit4CoreRunner.java:25)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4Launcher.execute(JUnit4Launcher.java:30)\r\n\tat com.microsoft.java.test.runner.Launcher.main(Launcher.java:53)\r\n"}}-TestRunner>.
[ERROR] 
.
[ERROR] @@<TestRunner-{"name":"error", "attributes":{"message":"Failed to run Junit tests", "details":"java.lang.NullPointerException\r\n\tat com.microsoft.java.test.runner.common.TestMessageItem.getStacktrace(TestMessageItem.java:44)\r\n\tat com.microsoft.java.test.runner.common.TestMessageItem.<init>(TestMessageItem.java:39)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4RunnerUtil.getRequestForClasses(JUnit4RunnerUtil.java:91)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4RunnerUtil.createTestReferences(JUnit4RunnerUtil.java:45)\r\n\tat com.microsoft.java.test.runner.junit4.CustomizedJUnit4CoreRunner.run(CustomizedJUnit4CoreRunner.java:25)\r\n\tat com.microsoft.java.test.runner.junit4.JUnit4Launcher.execute(JUnit4Launcher.java:30)\r\n\tat com.microsoft.java.test.runner.Launcher.main(Launcher.java:53)\r\n"}}-TestRunner>.
[ERROR] 
.

```

--- a/java-extension/com.microsoft.java.test.runner/src/main/java/com/microsoft/java/test/runner/testng/TestNGLauncher.java
+++ b/java-extension/com.microsoft.java.test.runner/src/main/java/com/microsoft/java/test/runner/testng/TestNGLauncher.java
@@ -50,11 +50,11 @@ public class TestNGLauncher implements ITestLauncher {
     }
 
     private static String getClassName(String clazz) throws ClassNotFoundException {
-        return Class.forName(clazz).getCanonicalName();
+        return Class.forName(clazz).getName();
     }
 
     private static String getClassNameFromMethod(String clazz) throws ClassNotFoundException {
-        return Class.forName(clazz.substring(0, clazz.lastIndexOf("#"))).getCanonicalName();
+        return Class.forName(clazz.substring(0, clazz.lastIndexOf("#"))).getName();
     }
 
     private static String getMethodName(String clazz) {

